### PR TITLE
Mapsite

### PR DESCRIPTION
**Pull request by Shimogawa**
有趣的是，即使在−100°C 的低温下，19F核磁共振谱也无法分辨轴向的氟原子与水平面的氟原子，只有一个简单的氟共振峰；这表明在毫秒单位级时，五氟化磷的五个氟原子是等同的，轴向与水平方向的氟原子发生着极快的交换作用。电子衍射和单晶X射线衍射没有发现这种影响，是因为他们的时间单位比核磁共振谱学更短，从而得以分辨出两种不同的氟原子。

这个现象首先由 Gutowsky 注意到，目前一般通过Berry假旋转作用来解释，即认为两个三角双锥结构通过变为一个四方锥构型的中间体，在不停地发生着转换。

### CODE REVIEW

**Review by chuigda — APPROVED** [review]
The pull request message is very funny, but I prefer not talking about politics or current events here, though.

File: backstage/src/executor/map_site.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+   const newRight: GeneratedBranch = {
+      desc,
+      next: newRightSite
+   }
+   curr.right = newRight
+}
+
+const MAX_DEPTH = 4
+
+// TODO: use iterative, avoid recursion
```
Comment (chuigda) [inline backstage/src/executor/map_site.ts:85]: A recursion of depth 4 is completely acceptable, no need to mark TODO here.

File: backstage/src/executor/map_site.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      this.site = site
+   }
+
+   isLeafSite() {
+      return this.left === undefined
+   }
+}
+
+export class GeneratedBranch {
+   desc: string
```
Comment (chuigda) [inline backstage/src/executor/map_site.ts:25]: use `MaybeTranslationKey` or `MaybeTranslatable` instead (the latter supports some "string template" semantics)

**Comment by CousinZe** [comment]
> The pull request message is very funny, but I prefer not talking about politics or current events here, though.

Then what's this?

https://github.com/Pr47/xjbutil/blob/master/src/liberty.rs#L15-L43

File: backstage/src/executor/map_site.ts (hunk 2 above)
Comment (Shimogawa) [reply backstage/src/executor/map_site.ts:25]: after compilation?

**Comment by Shimogawa** [comment]
> The pull request message is very funny, but I prefer not talking about politics or current events here, though.

random news from wikipedia

**Comment by chuigda** [comment]
> > The pull request message is very funny, but I prefer not talking about politics or current events here, though.
> 
> random news from wikipedia

Sure, no more news, just poetry or lyrics. Also you may use hitokoto.com

File: backstage/src/executor/map_site.ts (hunk 2 above)
Comment (chuigda) [reply backstage/src/executor/map_site.ts:25]: Sure.

**Review by CousinZe — APPROVED** [review]
Nice work. It would KEAL.

**Pull request merged**